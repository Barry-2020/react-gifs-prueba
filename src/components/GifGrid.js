import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, seImages] = useState([])
    const { data:images, loading } = useFetchGifs( category );



    // encodeURI reemplaza espacios formato url
    return (
        <>
        <h3 className='animate__animated animate__fadeInLeft'>{category}</h3>
        { loading && <p className='animate__animated animate__flash'>Loading...</p>}
        <div className='card-grid'>
                {
                    images.map(( img ) =>(
                        <GifGridItem 
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
   
        </div>
        </>
        
    )
}
