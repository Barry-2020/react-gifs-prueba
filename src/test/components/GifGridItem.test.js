import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en Gif Gird', () => {

    const title = "Un titulo";
    const url = 'http://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );
   

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('debe tener una img igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log( img.prop('src') ); 

        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );

    })

    test('debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const vale = div.props().className;
        // console.log( div.props().className.indexOf('animate__fadeInLeft') ); 
        
        expect( vale.includes('animate__fadeIn') ).toBe( true );
    })

 })