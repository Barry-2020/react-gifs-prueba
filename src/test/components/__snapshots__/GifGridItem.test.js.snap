// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Gif Gird debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
